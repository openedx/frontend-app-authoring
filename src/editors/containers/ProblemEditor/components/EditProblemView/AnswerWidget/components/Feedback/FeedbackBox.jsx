import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import { answerOptionProps } from '../../../../../../../data/services/cms/types';
import FeedbackControl from './FeedbackControl';
import messages from './messages';
import { ProblemTypeKeys } from '../../../../../../../data/constants/problem';

export const FeedbackBox = ({
  answer,
  problemType,
  setSelectedFeedback,
  setUnselectedFeedback,
  images,
  isLibrary,
  learningContextId,
  // injected
  intl,
}) => {
  const props = {
    answer,
    intl,
    images,
    isLibrary,
    learningContextId,
  };

  return ((problemType === ProblemTypeKeys.MULTISELECT) ? (
    <div className="bg-light-300 p-4 mt-3 rounded text-primary-500">
      <FeedbackControl
        key={`selectedfeedback-${answer.id}`}
        feedback={answer.selectedFeedback}
        labelMessage={messages.selectedFeedbackLabel}
        labelMessageBoldUnderline={messages.selectedFeedbackLabelBoldUnderlineText}
        onChange={setSelectedFeedback}
        type="selected"
        {...props}
      />
      <FeedbackControl
        key={`unselectedfeedback-${answer.id}`}
        feedback={answer.unselectedFeedback}
        labelMessage={messages.unSelectedFeedbackLabel}
        labelMessageBoldUnderline={messages.unSelectedFeedbackLabelBoldUnderlineText}
        onChange={setUnselectedFeedback}
        type="unselected"
        {...props}
      />
    </div>
  ) : (
    <div className="bg-light-300 p-4 mt-3 rounded text-primary-500">
      <FeedbackControl
        key={`selectedfeedback-${answer.id}`}
        feedback={answer.selectedFeedback}
        labelMessage={messages.selectedFeedbackLabel}
        labelMessageBoldUnderline={messages.selectedFeedbackLabelBoldUnderlineText}
        onChange={setSelectedFeedback}
        type="selected"
        {...props}
      />
    </div>
  ));
};
FeedbackBox.propTypes = {
  answer: answerOptionProps.isRequired,
  problemType: PropTypes.string.isRequired,
  setAnswer: PropTypes.func.isRequired,
  setSelectedFeedback: PropTypes.func.isRequired,
  setUnselectedFeedback: PropTypes.func.isRequired,
  images: PropTypes.shape({}).isRequired,
  learningContextId: PropTypes.string.isRequired,
  isLibrary: PropTypes.bool.isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(FeedbackBox);
