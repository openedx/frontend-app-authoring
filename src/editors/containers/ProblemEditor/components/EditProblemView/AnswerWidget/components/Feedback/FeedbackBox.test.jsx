import { shallow } from '@edx/react-unit-test-utils';
import { FeedbackBox } from './FeedbackBox';

const answerWithFeedback = {
  id: 'A',
  title: 'Answer 1',
  correct: true,
  selectedFeedback: 'some feedback',
  unselectedFeedback: 'unselectedFeedback',
  problemType: 'sOMepRObleM',
  images: {},
  isLibrary: false,
  learningContextId: 'course+org+run',
};

const props = {
  answer: answerWithFeedback,
  intl: {},
};

describe('FeedbackBox component', () => {
  test('renders as expected with default props', () => {
    expect(shallow(<FeedbackBox {...props} />).snapshot).toMatchSnapshot();
  });
  test('renders as expected with a numeric input problem', () => {
    expect(shallow(<FeedbackBox {...props} problemType="numericalresponse" />).snapshot).toMatchSnapshot();
  });
  test('renders as expected with a multi select problem', () => {
    expect(shallow(<FeedbackBox {...props} problemType="choiceresponse" />).snapshot).toMatchSnapshot();
  });
});
