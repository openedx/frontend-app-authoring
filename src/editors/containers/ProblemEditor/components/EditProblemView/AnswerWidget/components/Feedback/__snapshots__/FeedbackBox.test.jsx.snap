// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackBox component renders as expected with a multi select problem 1`] = `
<div
  className="bg-light-300 p-4 mt-3 rounded text-primary-500"
>
  <FeedbackControl
    answer={
      {
        "correct": true,
        "id": "A",
        "images": {},
        "isLibrary": false,
        "learningContextId": "course+org+run",
        "problemType": "sOMepRObleM",
        "selectedFeedback": "some feedback",
        "title": "Answer 1",
        "unselectedFeedback": "unselectedFeedback",
      }
    }
    feedback="some feedback"
    intl={{}}
    key="selectedfeedback-A"
    labelMessage={
      {
        "defaultMessage": "Show following feedback when {answerId} {boldunderline}:",
        "description": "Label text for feedback if option is selected",
        "id": "authoring.answerwidget.feedback.selected.label",
      }
    }
    labelMessageBoldUnderline={
      {
        "defaultMessage": "is selected",
        "description": "Bold & underlined text for feedback if option is selected",
        "id": "authoring.answerwidget.feedback.selected.label.boldunderline",
      }
    }
    type="selected"
  />
  <FeedbackControl
    answer={
      {
        "correct": true,
        "id": "A",
        "images": {},
        "isLibrary": false,
        "learningContextId": "course+org+run",
        "problemType": "sOMepRObleM",
        "selectedFeedback": "some feedback",
        "title": "Answer 1",
        "unselectedFeedback": "unselectedFeedback",
      }
    }
    feedback="unselectedFeedback"
    intl={{}}
    key="unselectedfeedback-A"
    labelMessage={
      {
        "defaultMessage": "Show following feedback when {answerId} {boldunderline}:",
        "description": "Label text for feedback if option is not selected",
        "id": "authoring.answerwidget.feedback.unselected.label",
      }
    }
    labelMessageBoldUnderline={
      {
        "defaultMessage": "is not selected",
        "description": "Bold & underlined text for feedback if option is not selected",
        "id": "authoring.answerwidget.feedback.unselected.label.boldunderline",
      }
    }
    type="unselected"
  />
</div>
`;

exports[`FeedbackBox component renders as expected with a numeric input problem 1`] = `
<div
  className="bg-light-300 p-4 mt-3 rounded text-primary-500"
>
  <FeedbackControl
    answer={
      {
        "correct": true,
        "id": "A",
        "images": {},
        "isLibrary": false,
        "learningContextId": "course+org+run",
        "problemType": "sOMepRObleM",
        "selectedFeedback": "some feedback",
        "title": "Answer 1",
        "unselectedFeedback": "unselectedFeedback",
      }
    }
    feedback="some feedback"
    intl={{}}
    key="selectedfeedback-A"
    labelMessage={
      {
        "defaultMessage": "Show following feedback when {answerId} {boldunderline}:",
        "description": "Label text for feedback if option is selected",
        "id": "authoring.answerwidget.feedback.selected.label",
      }
    }
    labelMessageBoldUnderline={
      {
        "defaultMessage": "is selected",
        "description": "Bold & underlined text for feedback if option is selected",
        "id": "authoring.answerwidget.feedback.selected.label.boldunderline",
      }
    }
    type="selected"
  />
</div>
`;

exports[`FeedbackBox component renders as expected with default props 1`] = `
<div
  className="bg-light-300 p-4 mt-3 rounded text-primary-500"
>
  <FeedbackControl
    answer={
      {
        "correct": true,
        "id": "A",
        "images": {},
        "isLibrary": false,
        "learningContextId": "course+org+run",
        "problemType": "sOMepRObleM",
        "selectedFeedback": "some feedback",
        "title": "Answer 1",
        "unselectedFeedback": "unselectedFeedback",
      }
    }
    feedback="some feedback"
    intl={{}}
    key="selectedfeedback-A"
    labelMessage={
      {
        "defaultMessage": "Show following feedback when {answerId} {boldunderline}:",
        "description": "Label text for feedback if option is selected",
        "id": "authoring.answerwidget.feedback.selected.label",
      }
    }
    labelMessageBoldUnderline={
      {
        "defaultMessage": "is selected",
        "description": "Bold & underlined text for feedback if option is selected",
        "id": "authoring.answerwidget.feedback.selected.label.boldunderline",
      }
    }
    type="selected"
  />
</div>
`;
